From d861aab0b57fe0bf1c51a2aad2822650d6219aa4 Mon Sep 17 00:00:00 2001
From: Zhenbo Gao <zhenbo.gao@windriver.com>
Date: Tue, 21 Aug 2018 06:19:19 +0000
Subject: [PATCH 005/176] Revert "tipc: fix use-after-free"

This reverts commit c549de482f89043eddda3ea33075180e8c479e49.

A big patch set for upgrading TIPC(to v4.18) will come soon

Signed-off-by: Zhenbo Gao <zhenbo.gao@windriver.com>
---
 net/tipc/netlink_compat.c |    6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/net/tipc/netlink_compat.c b/net/tipc/netlink_compat.c
index 750949d..9bfe886 100644
--- a/net/tipc/netlink_compat.c
+++ b/net/tipc/netlink_compat.c
@@ -258,15 +258,13 @@ static int tipc_nl_compat_dumpit(struct tipc_nl_compat_cmd_dump *cmd,
 	arg = nlmsg_new(0, GFP_KERNEL);
 	if (!arg) {
 		kfree_skb(msg->rep);
-		msg->rep = NULL;
 		return -ENOMEM;
 	}
 
 	err = __tipc_nl_compat_dumpit(cmd, msg, arg);
-	if (err) {
+	if (err)
 		kfree_skb(msg->rep);
-		msg->rep = NULL;
-	}
+
 	kfree_skb(arg);
 
 	return err;
-- 
1.7.9.5

